### PR TITLE
EPD-864 Collect revision usage during test case run

### PR DESCRIPTION
**Pull request by nicolewhite**

### CODE REVIEW

File: autoblocks/_impl/context_vars.py [hunk 1]
```diff
@@ -1,14 +1,67 @@
...
+
+def register_test_case_revision_usage(
+    entity_id: str,
+    entity_type: RevisionType,
+    revision_id: str,
+) -> None:
+    ctx = test_case_run_context_var.get()
+    if ctx is None:
+        # Shouldn't happen, but just return
+        return
```
Comment (dmorton2297) [inline autoblocks/_impl/context_vars.py:53]: Maybe return an empty list so you don't have to do check for None

```
testCaseRevisionUsage=[usage.serialize() for usage in revision_usage] if revision_usage else None,
```

in run.py

File: tests/autoblocks/test_run_test_suite.py [hunk 2]
```diff
@@ -1743,3 +1776,173 @@ def test_run_stops_if_start_fails(httpx_mock):
...
+            testCaseOutput="z!",
+            testCaseDurationMs=ANY_NUMBER,
+            testCaseRevisionUsage=[
+                dict(
+                    entityExternalId="prompt-a",
+                    entityType="prompt",
+                    revisionId="prompt-a-mock-revision-id",
+                    usedAt=mock.ANY,
+                ),
+            ],
```
Comment (dmorton2297) [inline tests/autoblocks/test_run_test_suite.py:1927]: Nice!

**Review by dmorton2297 — APPROVED** [review]

File: autoblocks/_impl/context_vars.py (hunk 1 above)
Comment (nicolewhite) [reply autoblocks/_impl/context_vars.py:53]: i think i want it to be None rather than an empty list when we send it to our API

Comment (nicolewhite) [reply autoblocks/_impl/context_vars.py:53]: also this is register_test_case_revision_usage not get_test_case_revision_usage

**Pull request merged**